iml-agent/src/action_plugins/lamigo.rs

```diff
@@ -0,0 +1,175 @@
+             -u {user} \
+             -s {hot_pool} \
+             -t {cold_pool} \
+             -a {min_age} \
+             {mount_point}\n\
+             ",
+            fs = self.fs,
+            mdt = self.mdt,
+            cold_pool = self.cold_pool,
+            hot_pool = self.hot_pool,
+            min_age = self.min_age,
+            user = self.user,
+            mount_point = self.mount_point,
+        )
+    }
+}
+
+pub async fn create_lamigo_service_unit(c: Config) -> Result<(), ImlAgentError> {
+    if !c.force {
+        let is_mounted = is_filesystem_mounted(&c.mount_point).await?;
+        if !is_mounted {
+            mount_filesystem(&c).await?;
+        }
+    }
+    create_lamigo_service_unit_internal(SYSTEMD_DIR, &c)
+        .err_into()
+        .await
+}
+
+pub async fn mount_filesystem(c: &Config) -> std::io::Result<()> {
+    // according to http://wiki.lustre.org/Mounting_a_Lustre_File_System_on_Client_Nodes
+    // we need to execute mount command
+    //   mount -t lustre \
+    //     [-o <options> ] \
+    //     <MGS NID>[:<MGS NID>]:/<fsname> \
+    //     /lustre/<fsname>
+    let mut command = Command::new("mount");
+    let process: Child = command
+        .arg("-t")
+        .arg("lustre")
+        .arg(c.lustre_device.clone())
+        .arg(c.mount_point.clone())
+        .spawn()?;
+    let output: Output = process.wait_with_output().await?;
+    match output.status.success() {
+        true => Ok(()),
+        false => Err(io::Error::new(
+            io::ErrorKind::Other,
+            format!("Cannot execute command {:?}", command),
+        )),
+    }
+}
+
+pub async fn is_filesystem_mounted(mount_point: &str) -> std::io::Result<bool> {
+    // `mountpoint -q /mnt/something` produces no output,
+    // but signals the caller with the exit code
+    let process: Child = Command::new("mountpoint")
+        .arg("-q")
+        .arg(mount_point)
+        .spawn()?;
+    let output: Output = process.wait_with_output().await?;
+    Ok(output.status.success())
+}
+
+async fn create_lamigo_service_unit_internal<P: AsRef<Path>>(
+    dir: P,
+    c: &Config,
+) -> std::io::Result<()> {
+    fs::create_dir_all(&dir).await?;
+    let file = dir
+        .as_ref()
+        .join(format!("lamigo-{}-{}.service", c.fs, c.mdt));
+    let cnt = format!("{}", c);
+    fs::write(file, cnt.as_bytes()).await
+}
+
+#[cfg(test)]
+mod lamigo_tests {
+    use super::*;
+    use insta::assert_display_snapshot;
+    use tempfile::tempdir;
+
+    #[tokio::test]
+    async fn test_works() {
+        // cargo run --package iml-agent --bin iml-agent -- lamigo
+        // --cold_pool COLD_POOL --hot_pool HOT_POOL
+        // --fs LU_TEST --lustre_device 192.168.0.100@tcp0:/spfs
+        // --mdt MDT000 --min_age 10000 --user nick --mount_point /mnt/lustre
+        let config = Config {
+            fs: "lu_test".into(),
+            mdt: "lustre-MDT0000".into(),
+            user: "cl1".into(),
+            min_age: 35353,
```

Comment:
`min_age` in the comment above says `10000`, why the difference?

Reply:
Agree, the comments and the code must correspond.

Reply:
> Agree, the comments and the code must correspond.

Or comments should not exist ;-)